index.adoc

```diff
@@ -602,6 +602,38 @@ MUST NOT perform type validation on any value in the array before submitting the
 
 Use of this member is OPTIONAL.
 
+A non-normative example
+
+[source,json]
+----
+{
+  "href": "https://example.io/persons", "rel": ["create-form"], "method": "POST",
+  "value": [
+    {
+      "name": "friends", "type": "object", "form": {
+        "type": "set", "itemtype": "object",
+        "value": [
+          { "name": "firstName" },
+          { "name": "email" }
+        ]
+      }
+    }
+  ]
+}
+----
```

Comment:
If the goal is to allow a submission of a set of friend objects, IIUC, this would be the linked form definition based on the current spec:

```json
{
  "href": "https://example.io/persons", "rel": ["create-form"], "method": "POST",
  "value": [
    { "name": "friends", "type": "set", "itemtype": "object", 
      "form": { 
        "value": [
          { "name": "firstName" },
          { "name": "email" }
        ]
      }
    }
  ]
}
```

The `friends` field is of type `set`, and each item within the set is an object, which is why there should be `"itemtype": "object"`.

The thing I think needs to be clarified in the spec is, should we still use the `form` field member, which currently has only been described in the context of `"type":"object"` or if we introduce a new field member named `itemform` which only applies if the field `type` is `set` or `array` *and* the `itemtype` member equals `object`.

The benefit of re-using `form` for set/array items is that we don't need to introduce a new member.  The benefit of introducing an `itemform` member is that it feels more accurate and self-documenting.  If `itemform` were added to the spec, the above would be this instead:

```json
{
  "href": "https://example.io/persons", "rel": ["create-form"], "method": "POST",
  "value": [
    { "name": "friends", "type": "set", "itemtype": "object", 
      "itemform": { 
        "value": [
          { "name": "firstName" },
          { "name": "email" }
        ]
      }
    }
  ]
}
```

In this scenario, a `form` member within a form field would *only* be inspected if the `type` is `object` and ignored otherwise, and an `itemform` member would *only* be inspected if the `itemtype` is `object`.

Any thoughts?